src/main/kotlin/no/nav/medlemskap/regler/v1/ReglerForLovvalg.kt

```diff
@@ -90,22 +98,42 @@ class ReglerForLovvalg(val datagrunnlag: Datagrunnlag) : Regler() {
             else -> nei("Brukeren er ikke norsk statsborger")
         }
     }
+    private fun sjekkAdresseType(): Resultat {
+
+        val adresser = arrayListOf<Adresse>()
+        adresser.addAll(postadresser)
+        adresser.addAll(bostedsadresse)
+        adresser.addAll(midlertidigAdresse)
+
 
-    private fun sjekkLandskode(): Resultat =
+        val adressetyper = adresser.hentAdressetyperIKontrollperiode(kontrollPeriodeForPersonhistorikk)
+
+        return when {
+            adressetyper alleEr Adressetype.BOAD.name -> ja()
+            adressetyper inneholder Adressetype.BOAD.name && adressetyper inneholder Adressetype.POST.name -> ja()
```

Comment:
Jeg sjekket tabellen for adressetype kombinasjoner på denne siden: https://confluence.adeo.no/pages/viewpage.action?pageId=358576589
Og det ser ut som om du har glemt å tenke på nr. 5. Kombinasjonen BOAD og POST skal bare aksepteres så lenge det ikke også er med en UTAD

Reply:
Burde det kanskje generelt vært en sjekk i hvert av tilfellene her, på at det ikke er noen andre adresser i listen enn de du ønsker deg? At `inneholder` komplementeres med `inneholderIkke` eller kanskje `inneholderAntall`, for å verifisere at de du sjekker om den inneholder er alt?